### PR TITLE
Raster layer can use XML extent like vector layer

### PR DESCRIPTION
**Pull request by rldhont**
## Description

The trust flag at the project level is only used to read vector layer extent from xml, not from provider.

This flag was not available for raster layer.

We propose to add the same capabilities in raster layer

### CODE REVIEW

**Comment by m-kuhn** [comment]
Could we just move it up to QgsMapLayer?

File: src/core/raster/qgsrasterlayer.h [hunk 1]
```diff
@@ -121,6 +123,14 @@ class CORE_EXPORT QgsRasterLayer : public QgsMapLayer
        */
       QgsCoordinateTransformContext transformContext = QgsCoordinateTransformContext();
 
+
+      /**
+       * If TRUE, the layer extent will be read from XML (i.e. stored in the
+       * project file). If FALSE, the extent will be determined by the provider on layer load.
+       * \since QGIS 3.16
+       */
+      bool readExtentFromXml = false;
```
Comment (nyalldawson) [inline src/core/raster/qgsrasterlayer.h:132]: Can we use flags here instead? I'm not a fan of adding setReadExtentFromXml/readExtentFromXml to the already complex layer api

**Comment by rldhont** [comment]
@m-kuhn and @nyalldawson the idea was in a first implementation to mimic the vector layer use xml extent for raster. It is to speed up QGIS Server project load.

I have planed to implement a way to manage the XML extent and to extend this capabilities to other layer types.

**Comment by m-kuhn** [comment]
Can you elaborate on the advantage of one implementation per layer type over one single implementation? Couldn't we avoid code duplication and maintenance overhead by having one single implementation?

**Comment by nyalldawson** [comment]
I'd rather see a clean API added going forward vs spreading the "messier" API to more classes...

**Comment by rldhont** [comment]
With https://github.com/qgis/QGIS/pull/38464 merged, it will be easier to propose e cleaner API.

**Pull request closed without merging**